library/HTMLPurifier/ElementDef.php

```diff
@@ -176,7 +176,7 @@ public function mergeIn($def)
 
         if (!empty($def->content_model)) {
             $this->content_model =
-                str_replace("#SUPER", $this->content_model, $def->content_model);
+                str_replace("#SUPER", (string)$this->content_model, $def->content_model);
```

Comment:
This smells like a (possibly benign) bug. The documentation claims that content_model is always a string, so maybe if it is sometimes null we were supposed to not attempt to do this replacement in that case (maybe #SUPER is never set in those cases)

Reply:
Did a quick static analysis: A non-string value for $this->content_model originates from: https://github.com/ezyang/htmlpurifier/blob/master/library/HTMLPurifier/HTMLModule.php#L215
(null is passed to the create method here: https://github.com/ezyang/htmlpurifier/blob/master/library/HTMLPurifier/HTMLModule.php#L157)